### PR TITLE
feature(whl): add single loss landscape visualizer

### PR DESCRIPTION
**Pull request by kxzxvbk**

### CODE REVIEW

File: fling/utils/visualize_utils/demo/demo_hessian_eigen_value.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    dataset = DATASET_REGISTRY.build('cifar10', dataset_config, train=False)
+
+    # Test dataset is for generating loss landscape.
+    test_dataset = [dataset[i] for i in range(100)]
+    test_dataloader = DataLoader(test_dataset, batch_size=100)
+
+    # Step 2: prepare the model.
+    model = ToyModel()
+
+    # Step 3: train the randomly initialized model.
```
Comment (KyeGuo) [inline fling/utils/visualize_utils/demo/demo_hessian_eigen_value.py:65]: Train a randomly initialized model or load a pre-trained model defined within the fling framework from local:
```
from easydict import EasyDict
from fling.utils.registry_utils import MODEL_REGISTRY

model_arg=EasyDict(dict(
    name='resnet8',
    input_channel=3,
    class_number=10,
))
model_name = model_arg.pop('name')
model = MODEL_REGISTRY.build(model_name, **model_arg)
model.load_state_dict(torch.load('./local_model.ckpt'))
```

File: fling/utils/visualize_utils/demo/demo_hessian_eigen_value.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            loss = criterion(pred_y, data_y)
+            optimizer.zero_grad()
+            loss.backward()
+            optimizer.step()
+    model.to('cpu')
+
+    # Step 4: plot the loss landscape after training the model.
+    # Only one line of code for visualization!
+    res = calculate_hessian_dominant_eigen_values(model, iter_num=20, dataloader=test_dataloader, device='cuda')
+    print(res)
```
Comment (KyeGuo) [inline fling/utils/visualize_utils/demo/demo_hessian_eigen_value.py:86]: When using it, I save the corresponding hessian eigen values for each layer to a csv file:
```
import csv

store_path = './hessian_eigen_value.csv'
with open(store_path, 'w', newline='', encoding='utf-8') as csv_file:
    writer = csv.writer(csv_file)
    writer.writerow(['Layers', name])
    for key, value in res.items():
        writer.writerow([key, value])
```

File: fling/utils/visualize_utils/demo/demo_single_loss_landscape.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    dataset = DATASET_REGISTRY.build('cifar10', dataset_config, train=False)
+
+    # Test dataset is for generating loss landscape.
+    test_dataset = [dataset[i] for i in range(100)]
+    test_dataloader = DataLoader(test_dataset, batch_size=100)
+
+    # Step 2: prepare the model.
+    model = ToyModel()
+
+    # Step 3: train the randomly initialized model.
```
Comment (KyeGuo) [inline fling/utils/visualize_utils/demo/demo_single_loss_landscape.py:65]: To load a pre-trained model defined within the fling framework from local:
```
from easydict import EasyDict
from fling.utils.registry_utils import MODEL_REGISTRY

model_arg=EasyDict(dict(
    name='resnet8',
    input_channel=3,
    class_number=10,
))
model_name = model_arg.pop('name')
model = MODEL_REGISTRY.build(model_name, **model_arg)
model.load_state_dict(torch.load('./local_model.ckpt'))
```

File: fling/utils/visualize_utils/demo/demo_conv_kernel_visualize.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+from torchvision.models import resnet18
+
+from fling.utils import Logger
+from fling.utils.visualize_utils import plot_conv_kernels
+
+if __name__ == '__main__':
+    # Step 1: prepare the model.
+    model = resnet18(pretrained=True)
```
Comment (KyeGuo) [inline fling/utils/visualize_utils/demo/demo_conv_kernel_visualize.py]: To load a model defined within the fling framework from local, the following code can be use:
```
from easydict import EasyDict
from fling.utils.registry_utils import MODEL_REGISTRY

# Step 1: prepare the model.
model_arg=EasyDict(dict(
    name='resnet8',
    input_channel=3,
    class_number=100,
))
model_name = model_arg.pop('name')
model = MODEL_REGISTRY.build(model_name, **model_arg)
model.load_state_dict(torch.load('./local_model.ckpt'))
```
